From 3eaf2b86be2399d5050124c45a27ab33ffdf6525 Mon Sep 17 00:00:00 2001
From: Beniamino Galvani <bgalvani@redhat.com>
Date: Wed, 20 Feb 2019 10:17:27 +0100
Subject: [PATCH] wireless-security: 'unknown' method requires non-empty
 secrets

secrets_hints can be a empty array when there are no hints.  See also
similar commit [1] for nmcli secret agent.

[1] https://gitlab.freedesktop.org/NetworkManager/NetworkManager/commit/b32c03c879603ad6a8d53b083a932a9d525c0253

Fixes: 0173934c831fac41d1203affd1fdf3d7fffeb22e

https://gitlab.gnome.org/GNOME/network-manager-applet/merge_requests/39
---
 src/wireless-security/wireless-security.c | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/src/wireless-security/wireless-security.c b/src/wireless-security/wireless-security.c
index 257675c1..d1b69c22 100644
--- a/src/wireless-security/wireless-security.c
+++ b/src/wireless-security/wireless-security.c
@@ -499,7 +499,7 @@ ws_802_1x_auth_combo_init (WirelessSecurity *sec,
 		active = item;
 	item++;
 
-	if (secrets_hints) {
+	if (secrets_hints && secrets_hints[0]) {
 		EAPMethodSimple *em_hints;
 
 		em_hints = eap_method_simple_new (sec, connection, EAP_METHOD_SIMPLE_TYPE_UNKNOWN,
-- 
2.20.1

